primitives/common/unused_type.go

```diff
@@ -53,21 +51,14 @@ func (ut *UnusedType) DefineSSZ(c *ssz.Codec) {
 
 // MarshalSSZ marshals the UnusedType object to SSZ format.
 func (ut *UnusedType) MarshalSSZ() ([]byte, error) {
-	return []byte{uint8(*ut)}, nil
+	buf := make([]byte, ssz.Size(ut))
+	return buf, ssz.EncodeToBytes(buf, ut)
 }
 
-func DecodeUnusedType(buf []byte, v *UnusedType) error {
-	// we special case construction of unused types, for efficiency
-	if len(buf) != 1 {
-		return fmt.Errorf("expected 1 byte got %d", len(buf))
-	}
-	//#nosec:G701 // UnusedType is uint8 and byte is uint8.
-	*v = UnusedType(buf[0])
-	return nil
+func (ut *UnusedType) ValidateAfterDecodingSSZ() error {
+	return ut.EnforceUnused()
 }
 
-func (*UnusedType) ValidateAfterDecodingSSZ() error { return nil }
```

Comment:
this was probably a bug and I think calling `ut.EnforceUnused()` in `ValidateAfterDecodingSSZ` is the correct thing to do (although the bug would have only showed up if we marshalled UnusedTypes by themselves, not as a part of an object